### PR TITLE
G-API: Align limit flag behavior of G-API demos with OMZ

### PR DESCRIPTION
**Pull request by mpashchenkov**
Minor:
- Removed limit flag from ImagesCapture for IFD;
- Added help message for FLAG_limit for SCD;
- Changed name of CustomCaptureSource class for GestRec demo (common class has the same name);
- throw with extended messages for GestRec demo.

### CODE REVIEW

File: demos/interactive_face_detection_demo/cpp_gapi/main.cpp [hunk 1]
```diff
@@ -424,8 +424,7 @@ int main(int argc, char *argv[]) {
         std::unique_ptr<Presenter> presenter;
 
          /** Get information about frame **/
-        std::shared_ptr<ImagesCapture> cap = openImagesCapture(FLAGS_i, FLAGS_loop, 0,
-            FLAGS_limit);
+        std::shared_ptr<ImagesCapture> cap = openImagesCapture(FLAGS_i, FLAGS_loop);
```
Comment (Wovchena) [inline demos/interactive_face_detection_demo/cpp_gapi/main.cpp:427]: Why is limit removed?

**Comment by Wovchena** [comment]
Jenkins please retry a build

**Comment by mpashchenkov** [comment]
@Wovchena, some OMZ demos don't pass tests with segfaults. This PR contains minor cosmetic changes of G-API demos now. I can't fix red build.

**Comment by Wovchena** [comment]
Yes, https://github.com/openvinotoolkit/open_model_zoo/pull/3236 broke this

**Comment by vladimir-dudnik** [comment]
ignore non relevant issues

**Pull request merged**